### PR TITLE
Backport 5.1.1

### PR DESCRIPTION
**Pull request by rubenwardy**
This attempts to reduce risk by balancing the complexity of the fix and the value of the fix.

Included fixes:

* #9114 - MAJOR - fixes networking performance
* #9133 - MAJOR - fixes crash
* a485d00 - Without this, `formspec_version[]` is broken. This is a trivial fix
* #9064 - Trivial build fix
* 8e63d22 - trivial
* #9239 - simple fix
* f6de86c - regression fix
* 706b6aa  - build fix

Not included:

* #9150 - probably high risk
* #9043
* 2907c0f Android doesn't work
* CSM fixes

* 37f771a95bd54e2072f07783df142da4a91e7bff

## How to test:

* Read and understand the issues of each fix
* Verify that the fix still works in this branch
  - [x] Major network
  - [x] Major `%` escaping
  - [x] Formspec version reset
  - [x] Build system fix for include
  - [x] Build fix for jsoncpp
  - [x] Trimming formspec_version
  - [x] Item eat on last item
  - [x] localplayer bound sound

### CODE REVIEW

**Comment by sfan5** [comment]
IMO the CSM fixes aren't important to have in 5.1.1.

Also a note: <b>DO NOT</b> merge this PR using the button on Github, follow the exact release instructions instead.

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,11 +17,11 @@ set(CLANG_MINIMUM_VERSION "3.4")
 # Also remember to set PROTOCOL_VERSION in network/networkprotocol.h when releasing
 set(VERSION_MAJOR 5)
 set(VERSION_MINOR 1)
-set(VERSION_PATCH 0)
+set(VERSION_PATCH 1)
 set(VERSION_EXTRA "" CACHE STRING "Stuff to append to version string")
 
 # Change to false for releases
-set(DEVELOPMENT_BUILD FALSE)
+set(DEVELOPMENT_BUILD TRUE)
```
Comment (sfan5) [inline CMakeLists.txt]: release commit is still missing

**Comment by paramat** [comment]
I must have missed something (i read dev channel logs), i thought we agreed to not do a point release? We are close to releasing 5.2.0 and there are no critical bugs (or are there? i cannot see any new critical bugs reported).

**Comment by rubenwardy** [comment]
There are two critical bugs here, and a few medium. Releasing 5.1.1 was discussed in the staff channel

**Comment by paramat** [comment]
Ok, those 2 are critical.
Unfortunately the staff channel has no logs, and many core devs may not be present as dev discussion is not expected there. Not a good place for a dev discussion =)

Maybe we should just release 5.2.0 now then? And avoid the extra complication. We have been wanting to release more often so this is a good way to do that. Only adds a week of feature freeze.

**Comment by nerzhul** [comment]
okay for a patch release, should be fine for end users awaiting for next release

**Comment by nerzhul** [comment]
as a side note, i think backporting and doing patch release monthly on the stable branch can be nice for our users to prevent waiting months

**Comment by paramat** [comment]
Hmm, releasing 5.2.0 now would seem rushed, so i am ok with tis patch release.
Strongly disapprove of monthly patch releases, far too much additional work and complexity, and usually unnecessary, only when we have critical bugs. We are obviously already overwhelmed.

File: CMakeLists.txt (hunk 1 above)
Comment (rubenwardy) [reply CMakeLists.txt]: This will be done when merging into stable-5

Comment (sfan5) [reply CMakeLists.txt]: Why?

Comment (rubenwardy) [reply CMakeLists.txt]: because rebasing everytime to move the release commit to the end is annoying

Comment (SmallJoker) [reply CMakeLists.txt]: `git rebase -i HEAD~10` is only annoying because it touches all files, hence rebuilding is needed.
Not an excuse IMO.

Comment (rubenwardy) [reply CMakeLists.txt]: it would be `git rebase -i HEAD~2` to reorder each time

**Comment by MoNTE48** [comment]
It’s probably worth adding my “little” Android PR if someone wants to update the game on Google Play...
If this is interesting, I will do another rebase and simplify it a bit.

**Pull request closed without merging**